### PR TITLE
[doc] updating the pipe balance doc a bit

### PR DESCRIPTION
**Pull request by min-xu-ai**
- Added some notes as I was using those two functions.
- Also added a warning to pipeline.py when the partition output is not supported.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: fairscale/nn/pipe/balance/__init__.py [hunk 1]
```diff
@@ -23,13 +23,20 @@
...
 
     pipe = Pipe(model, balance, chunks=8)
 
+.. note::
+    balance_by_time does not work with inplace ReLU because we exhausetively search
+    every partition boundary, which could hit an inplace ReLU.
+
+.. note::
+    If the model is larger than a single CUDA device memory, use "cpu"
+    in the balance_by_time function.
```
Comment (min-xu-ai) [inline fairscale/nn/pipe/balance/__init__.py:39]: Just a note, when I was using it on the 10B model, it crashed with a segfault. Obviously I had to use the cpu device. So I didn't get a useful split out of this function. In the end, my pipeline split has to be done by hand tuning anyway. :-(

File: fairscale/nn/pipe/pipeline.py [hunk 2]
```diff
@@ -180,7 +180,10 @@ def function(
             part_id: int = j,
         ) -> TensorOrTensors:
             with use_skip_tracker(skip_tracker), record_function("chunk%d-part%d" % (chunk_id, part_id)):
-                return partition(input)
+                ret = partition(input)
+                if type(ret) is list:
```
Comment (msbaines) [inline fairscale/nn/pipe/pipeline.py:184]: I'm not sure about this check. It will be run for every partition for every micro-batch.

**Review by msbaines — COMMENTED** [review]
Nice! I'm not sure about the warning though.

File: fairscale/nn/pipe/pipeline.py (hunk 2 above)
Comment (min-xu-ai) [reply fairscale/nn/pipe/pipeline.py:184]: Yeah, was worrying about that too. How about I turn it into a comment?

**Review by msbaines — APPROVED** [review]

**Pull request merged**